examples/adv-generic-train-loop.rs

```diff
@@ -0,0 +1,32 @@
+/// This advanced example shows how to work with dfdx in a generic
+/// training setting.
+use dfdx::prelude::{tensor_collection::TensorCollection, *};
+
+fn classification_train<
+    Inp: Shape,
+    Targ: Shape,
+    Model: ModuleMut<Tensor<Inp, E, D, OwnedTape<E, D>>, Output = Tensor<Targ, E, D, OwnedTape<E, D>>>
+        + TensorCollection<E, D>,
```

Comment:
Would love to simplify this to
```suggestion
    Inp,
    Targ,
    Model: ModuleMut<Inp, Output = Targ> + TensorCollection<E, D>,
```
So you don't need to worry about owned tapes/shapes/ or tensors even

Reply:
yes, this would probably better, i currently also have a big problem with: Model<Batched, Owned>, Model<Single, Owned>, Model<Batched, None>, Model<Single, None> to pass batched and single values/with(out) tape